File: data-label-api/src/main/java/com/btd/model/Annotation.java
package com.btd.model;

import java.io.Serializable;

import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;

@Entity
public class Annotation implements Serializable {
	private static final long serialVersionUID = 2121369575062788730L;

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Long id;
	private boolean principal;
	private int x;
	private int y;

	@ManyToOne(fetch = FetchType.EAGER)
	private Image image;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public boolean isPrincipal() {
		return principal;
	}

	public void setPrincipal(boolean principal) {
		this.principal = principal;
	}

	public int getX() {
		return x;
	}

	public void setX(int x) {
		this.x = x;
	}

	public int getY() {
		return y;
	}

	public void setY(int y) {
		this.y = y;
	}

	public Image getImage() {
		return image;
	}

	public void setImage(Image image) {
		this.image = image;
	}

}

File: data-label-web/src/app/model/Annotation.ts
import {Image} from './image';

export class Annotation {
  public id: number;
  public principal: boolean;
  public x: number;
  public y: number;
  public image: Image;

  constructor(x: number, y: number, principal: boolean) {
    this.x = x;
    this.y = y;
    this.principal = principal;
  }

  isMarked() {
    return ((this.x !== -1) && (this.y !== -1));
  }
}
